compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -1091,6 +1100,19 @@ impl ConstructorSet {
                     missing.push(NonExhaustive);
                 }
             }
+            ConstructorSet::Bool => {
+                let mut seen_bools = [false, false];
+                for b in seen.map(|ctor| ctor.as_bool().unwrap()) {
+                    seen_bools[b as usize] = true;
+                }
+                for b in [true, false] {
+                    if seen_bools[b as usize] {
+                        present.push(Bool(b));
+                    } else {
+                        missing.push(Bool(b));
+                    }
+                }
```

Comment:
This will be much simpler unrolled.

Reply:
Hahaha I wrote it unrolled and thought "this would be much cleaner with a loop" x) I agree unrolled is more legible

Reply:
Fixed